src/main/java/com/epam/page/object/generator/model/SearchRule.java

```diff
@@ -1,43 +1,43 @@
 package com.epam.page.object.generator.model;
 
+import static java.util.Arrays.asList;
+
 import com.google.common.collect.Lists;
 import java.io.IOException;
 import java.util.List;
-
 import org.json.simple.JSONObject;
 import org.jsoup.Jsoup;
 import org.jsoup.nodes.Document;
 import org.jsoup.nodes.Element;
 import org.jsoup.select.Elements;
 import us.codecraft.xsoup.Xsoup;
 
-import static java.util.Arrays.asList;
-
 public class SearchRule {
 
+    public String tag;
+    public String requiredAttribute;
+    public List<String> classes;
+    public Pairs attributes;
     private String uniqueness;
     private String title;
     private String type;
     private String css;
     private String xpath;
     private List<SearchRule> innerSearchRules;
 
-    public SearchRule() { }
-    public String tag;
-    public String requiredAttribute;
-    public List<String> classes;
-    public Pairs attributes;
+    public SearchRule() {
```

Comment:
is it valid to create an empty SearchRule?

Reply:
Yes, we're using it in tests.

Reply:
but if someone creates an empty rule not in tests, it's gonna break things. Isn't it?

Reply:
Why should it break something?

Reply:
So our code can hande empty rules without throwing any exceptions? NullPointers etcs

Reply:
It should get ValidationException, because null type is not supported

Reply:
If you declare public constructor, it becomes a part of SearchRule API. That means I can create empty search rule *programmatically*, which makes no sense. That's a potential cause of errors.

Reply:
Ok, should i fix it in this branch?